### DISCUSSION TITLE
Thread-safe에 대해 설명해주세요. (hint: critical section)

### DISCUSSION THREAD

**Discussion by eatingbug** [Team-Psyduck/tech-sharing#65] [General]
-

**Comment by eatingbug** [comment]
멀티스레드 프로그래밍에서 여러 스레드에서 특정 함수나 변수에 동시에 접근해도 프로그램 실행에 큰 문제가 없음을 의미한다.

**Comment by kkg5** [comment]
`Java`에서 `Thread-Safe` 한 코드를 작성하기 위한 방법으로
키워드 `synchronized`, 자료구조 `AtomicVariables`, `ConcurrentCollections` 등이 있습니다.

**Comment by SionBackEnd** [comment]
변경이 가능한 전역변수에 동시에 접근하여도 예측 가능한 결과값을 얻을 수 있음을 뜻합니다. 
Java에서 Thread-Safe한 코드를 잘못 작성할 경우 데드락 유발 및 성능 저하를 유발할 수 있습니다.
